koocook_core/views/comments.py

```diff
@@ -0,0 +1,47 @@
+from django.http import HttpRequest, JsonResponse
+from django.urls import resolve
+from django.views.decorators.http import require_http_methods
+from django.contrib.auth.decorators import login_required
+from ..models import Comment
+
+
+def get_all_comments_for(request: HttpRequest, item_id: int):
+    if request.method == 'GET':
+        current_url = resolve(request.path_info).route
+        if 'recipe' in current_url:
+            from ..models import Recipe
+            comments = Recipe.objects.get(pk=item_id).comment_set.all()
+        elif 'post' in current_url:
+            from ..models import Post
+            comments = Post.objects.get(pk=item_id).comment_set.all()
+        elif 'comment' in current_url:
+            comments = Comment.objects.get(pk=item_id).comment_set.all()
```

Comment:
@MaiMee1 The function is specifically registered to the particular routes. (which, in the case, is `recipes/<id>/comments`) So there's no chance of misapplication. Since there are only 3 items in the list, the dict would not be that helpful.

Reply:
ok then